data/service/raw-data-service.js

```diff
@@ -1118,6 +893,69 @@ exports.RawDataService = DataService.specialize(/** @lends RawDataService.protot
         }
     },
 
+    /**
+     * Map from a parent class to the mappings used by the service to 
+     * determine what subclass to create an instance of for a particular
+     * rawData object
+     * 
+     * For example, say a class 'Person' has 2 subclasses 'Employee' & 'Customer'. 
+     * RawDataService would evaluate each person rawData object against each item 
+     * in _rawDataTypeMappings and determine if that rawData should be an instance
+     * of 'Employee' or 'Customer'.
+     * @type {Map<ObjectDescpriptor:RawDataTypeMapping>}
+     */
+    
+    _descriptorToRawDataTypeMappings: {
+        value: undefined
+    },
+
+    /**
+     * Adds each mapping passed in to _descriptorToRawDataTypeMappings 
+     * 
+     * @method
+     * @param {Array<RawDataTypeMapping>} mappings 
+     */
+    _registerRawDataTypeMappings: {
+        value: function (mappings) {
+            var mapping, type, 
+                i, n;
+
+            for (i = 0, n = mappings ? mappings.length : 0; i < n; i++) {
+                mapping = mappings[i];
+                type = mapping.type;
```

Comment:
type itself isn't used, but type.parent is, 3 times, so that's probably what should be put in the local variable